packages/xod-client/src/project/selectors.js

```diff
@@ -70,9 +76,17 @@ export const getCurrentPatch = createSelector(
   (patchPath, project) => R.chain(XP.getPatchByPath(R.__, project), patchPath)
 );
 
+const projectHasSamePatchList = R.either(
+  (a, b) => a === b,
+  compareMapsBy(R.pipe(XP.listPatches, R.indexBy(XP.getPatchPath)))
+);
+
 export const getDeducedPinTypes = createMemoizedSelector(
   [getProject, getCurrentPatch],
-  [R.equals, R.equals],
+  [
+    projectHasSamePatchList,
```

Comment:
A dangerous presumption. Will work only if no external impact can affect used nodes exterior. Will stale if we have split view.